builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -1367,9 +1372,12 @@ func (vm *virtualMachine) createVirtualMachine(c *govmomi.Client) error {
 		}
 	} else {
 		if vm.useSDRS {
-			// FIXME: this is not returning a datastore ... will this
-			// still work?
-			datastore, err = finder.DatastoreCluster(context.TODO(), vm.datastore)
+			var pod *object.StoragePod
+			pod, err = finder.DatastoreCluster(context.TODO(), vm.datastore)
+			if err == nil {
+				sps := buildStoragePlacementSpecCreate(dcFolders, resourcePool, pod, configSpec)
```

Comment:
@markpeek how do we validate that this is a good approach?  `DatastoreCluster` is not used in any of the govmomi code.